### DISCUSSION TITLE
Can I transform a API REST list into multiple HTML elements in a Markdown Viewer with the {{#each }} tag?

### DISCUSSION THREAD

**Discussion by lollococce** [Budibase/budibase#4939] [Help] [answered]
Hi!

I have an API response from a REST datasource that is something like this thanks to a transformer

`[{ 'name': 'Name', 'id': 21223412, 'labels': [{'name': 'Label Name'}, {'name': 'Label Name 2'}, ...] }, {...}, {...}]`

Basically I was trying to show the data in the labels list as it's a Repeater using the Handlebars tags, but I dunno how the data source names are scoped.

So Am using a DP and a RP (Repeater) to show the API response, but then I don't wanna make another DP and RP nested to show labels, it's inefficient.
I tried to render the following code in a Markdown component for the labels list, but this is containing too many things, not just the labels list
`{{#each this }}<a href="">{{ this }}</a>{{/each}}`

What about `{{#each RP Name.labels }}<a href="">{{ this.name }}</a>{{/each}}`? this also does not work

How can I use the actual scoped labels list for the row? How do you give the naming for such a tag?

Thanks

**Comment by mjashanks** [comment, marked as answer]
Hey @lollococce 

There is not a great way to do this, I really think we should allow repeaters to access arrays, as you suggest. The only way that I can think of achieving this right now is using an embed component. 

It's exactly as you suggest about, except use an `Embed`. I tried it using the following javascript and it worked. The code below iterates a "topics" array and lists each "topic" inside a `<p> ... </p>`

```javascript
let html = "";

const topics = $("New Repeater.Get Single Repo.topics");

for(let topic of topics) {
	html += `<p>${topic}</p><br>`;
}

return html;
```
